Dockerfile

```diff
@@ -1,10 +1,22 @@
-FROM node:14
+FROM node:14 as builder
```

Comment:
Why do you need a separate builder?

Reply:
Multistage images can reduce the image size drastically and also reduce the layers count, which should be always welcome.

Reply:
Right, but how does it help specifically in this case?
The idea of the current layers is that the layers are ordered based on size / reason to change. So the code is in the bottom layer, because that is really small, and changes the most.

Whenever a bottom layer changes, while upper layers stay the same, only the updated layer needs to be downloaded.

Reply:
Maybe it's just a personal preference, but in my honest opinion, a container image should always be as small as possible. If you take this multistage approach, the image will not have an unnecessary installation layer, as well as other installation cache files.

Reply:
@marklagendijk Multi stage builds are awesome. It keeps everything from what you need at build time separate from the runtime. Look at the documentation at docker site and it will become clear. I use multi stage builds for anything I can.

https://docs.docker.com/build/building/multi-stage/